### PR TITLE
Backend Unit Tests

### LINKED ISSUE #95: Add Sprint 2 backend tests
Go back and test the code written in sprint 2 for the backend (mongodb models + nodejs/express api)

### PR DESCRIPTION
**Pull request by Arvind-Maan**
closes #95

### CODE REVIEW

File: server/src/routes/appointment.js [hunk 1]
```diff
@@ -32,52 +54,57 @@ router.post('/:buyer', (req, res) => {
     let provider = '';
 
     Service.findById(req.body.service_id).then(service => {
-        if (service) provider = service.provider;
-        else
+        if (service) {
```
Comment (RikiZimbakov) [inline server/src/routes/appointment.js:57]: this is the same code as right below just formatting, yes?

File: server/src/routes/services.js [hunk 2]
```diff
@@ -23,80 +24,106 @@ router.get('/', (req, res) => {
...
+                    .json({ errors: [{ service: 'services are empty' }] });
+            }
+        })
+        .catch(err => {
             return res
                 .status(404)
-                .json({ errors: [{ service: 'services are empty' }] });
-        }
-    });
+                .json({ errors: [{ service: 'service doesnt exist' }, err] });
```
Comment (RikiZimbakov) [inline server/src/routes/services.js:43]:
```suggestion
                .json({ errors: [{ service: 'service doesn't exist' }, err] }); 
```
its just spelling

File: server/src/routes/services.js [hunk 3]
```diff
@@ -23,80 +24,106 @@ router.get('/', (req, res) => {
...
+            .json({ errors: 'Cannot change user in service' });
     }
 
     //update the service based on req.body
     Service.findByIdAndUpdate(
         req.params.service_id,
         req.body,
         (err, service) => {
-            if (err) return res.status(500).json({ error: err });
+            if (err) return res.status(500).json({ errors: err });
```
Comment (RikiZimbakov) [inline server/src/routes/services.js:112]: why change this to errors?

File: server/src/routes/user.js [hunk 4]
```diff
@@ -141,13 +141,29 @@ router.get('/:username/services', (req, res) => {
...
+                                .status(200)
+                                .json({ errors: [{ error: err }] });
+                        });
+                } else
+                    return res
+                        .status(404)
+                        .json({ errors: [{ password: 'incorrect' }] });
+            })
+            .catch(err => {
+                return res.status(404).json({ error: err });
```
Comment (RikiZimbakov) [inline server/src/routes/user.js:162]: maybe you want errors here instead? just to stay consistent with the logic used in my above comment?
```suggestion
                return res.status(404).json({ errors: err });
```

**Review by RikiZimbakov — COMMENTED** [review]
i just have to run your code and check out my comments before merge please

File: server/src/routes/services.js (hunk 3 above)
Comment (Arvind-Maan) [reply server/src/routes/services.js:112]: errors is the correct key name, to be consistent with our responses everything should have errors : []

File: server/src/routes/user.js (hunk 4 above)
Comment (Arvind-Maan) [reply server/src/routes/user.js:162]: good catch

File: server/src/app.js [hunk 5]
```diff
@@ -1,13 +1,12 @@
...
-import cors from 'cors';
-import dotenv from 'dotenv';
-import express, { json, urlencoded } from 'express';
-import mongoose from 'mongoose';
-import logger from 'morgan';
-import indexRouter from './routes/index.js';
-import servicesRouter from './routes/services.js';
-import userRouter from './routes/user.js';
-import appointmentRouter from './routes/appointment.js';
+const cookieParser = require('cookie-parser');
```
[the remark this answers is not in the document]
Comment (Arvind-Maan) [reply server/src/app.js:1]: Mocha doesn't play nice with ES6 imports. Switching to CommonJS will be better for our backend. It's been very frustrating to deal with ES6 problems imo so I just decided to change it all

File: server/src/routes/appointment.js (hunk 1 above)
Comment (Arvind-Maan) [reply server/src/routes/appointment.js:57]: Not really formatting, more like restructuring. But yes

**Comment by Arvind-Maan** [comment]
> Looks good to me. Might want to restructure the test_utils of either the back end or the front end once the front end tests are done to have a more consistent folder structure, but that's for later.

since we use mocha in the backend and Jest in the frontend, the project structure will be different

**Pull request merged**